py-polars/polars/config.py

```diff
@@ -147,17 +150,21 @@ def __exit__(
         self._original_state = ""
 
     @classmethod
-    def load(cls, cfg: str) -> type[Config]:
+    def load(cls, cfg: Path | str) -> type[Config]:
         """
-        Load and set previously saved (or shared) Config options.
+        Load and set previously saved (or shared) Config options from json/file.
 
         Parameters
         ----------
         cfg : str
-            json string produced by ``Config.save()``.
+            json string produced by ``Config.save()``, or a filepath to the same.
 
         """
-        options = json.loads(cfg)
+        options = json.loads(
+            Path(normalise_filepath(cfg)).read_text()
+            if os.path.exists(cfg) or isinstance(cfg, Path)
```

Comment:
The `os.path.exists(cfg)` check seems problematic. This allows injection attacks where content is confused with file name. Say someone was loading config `{"foo": bar}`. This is a valid file name, so now I can create a file with this name in the current directory and have this line load arbitrary configs. Seems like it would be cleaner to have separate `load_from()` and `save_to()` methods which avoid confusion and have simpler type signatures.

Reply:
Hey @eltoder, good to see you here!
Totally missed this earlier - will take a look :)